### DISCUSSION TITLE
Storing secrets without deploying Vault/Credhub whatever

### DISCUSSION THREAD

**Discussion by Erwyn** [concourse/concourse#6772] [Help & Support] [answered]
Hi there,

I was wondering if there was a built-in way to store secrets without using external stores like the ones mentioned in the title. I'm just starting with concourse and I've seen https://concourse-ci.org/encryption.html but have a hard time figuring it out. I mean, how am I supposed to enter my secrets so that I can call them in the pipelines ?

Thanks !

**Comment by taylorsilva** [comment]
Welcome! That encryption article you linked is talking about encryption of info stored in the database. It has nothing to do with credential managers like vault.

There is no built-in secret manager. That's actually something we're trying to pull out of our codebase [prototypes](https://github.com/concourse/rfcs/blob/master/037-prototypes/proposal.md) and [var-sources](https://github.com/concourse/rfcs/blob/master/039-var-sources/proposal.md).

When you're setting a pipeline with `fly set-pipeline` you can pass in vars for local interpolation before sending it up to Concourse.

**Comment by Erwyn** [comment, marked as answer]
Hey,

Thanks for your answer! So if I get it correctly, although there isn't any
secret manager I can use the interpolation and thanks to encryption my
variables will be stored safely on concourse's side?

I understand this isn't a secret manager but it's still a way to store this
safely for a small deployment. I was considering deploying concourse for my
personal use but setting up a secret manager like vault alongside felt a
bit much...

Le lun. 29 mars 2021 à 21:35, Taylor Silva ***@***.***> a
écrit :

**Reply by taylorsilva** [reply]
You are correct! Should be fine for a personal deployment :)
